website/content/docs/agent/config/config-files.mdx

```diff
@@ -1586,15 +1586,98 @@ Valid time units are 'ns', 'us' (or 'µs'), 'ms', 's', 'm', 'h'."
 
 ## Raft Parameters
 
-- `raft_boltdb` ((#raft_boltdb)) This is a nested object that allows configuring
-  options for Raft's BoltDB based log store.
-
-  - `NoFreelistSync` ((#NoFreelistSync)) Setting this to `true` will disable
-    syncing the BoltDB freelist to disk within the raft.db file. Not syncing
-    the freelist to disk will reduce disk IO required for write operations
-    at the expense of potentially increasing start up time due to needing
-    to scan the db to discover where the free space resides within the file.
-
+- `raft_boltdb` ((#raft_boltdb)) **These fields are deprecated in Consul 1.15.0.
+  See the [`raft_logstore`](#raft_logstore) instead.** This is a nested
+  object that allows configuring options for Raft's BoltDB based log store.
+
+  - `NoFreelistSync` **This field is deprecated in Consul 1.15.0. See the
+    [`raft_logstore.boltdb.no_freelist_sync`](#raft_logstore_boltdb_no_freelist_sync) field
+    instead.** Setting this to `true` will disable syncing the BoltDB freelist
+    to disk within the raft.db file. Not syncing the freelist to disk will
+    reduce disk IO required for write operations at the expense of potentially
+    increasing start up time due to needing to scan the db to discover where the
+    free space resides within the file.
+
+- `raft_logstore` ((#raft_logstore)) This is a nested object that allows
+  configuring options for Raft's LogStore component which is used to persist
+  logs and crucial Raft state on disk during writes. This was added in Consul
+  1.15.
+
+  - `backend` ((#raft_logstore_backend)) This allows selection of which storage
+    engine to use to persist logs. Valid options are `boltdb` or `wal`. Default
+    is `boltdb`. As of Consul 1.15, `wal` is a new and experimental backend that
+    should be used with caution. See [our experimental WAL backend testing
+    guide](/consul/docs/upgrading/instructions/testing-experimental-raft-backend)
+    to learn how to safely evaluate it for your workload.
+
+  - `disable_log_cache` ((#raft_logstore_disable_log_cache)) This allows
+    disabling of the in-memory cache of recent logs. This exists mostly for
+    performance testing purposes. In theory the log cache prevents disk reads
+    for recent logs. In practice recent logs are still in OS page cache so tend
+    not to be slow to read using either backend. We recommend leaving it enabled
+    for now as we've not measured a significant improvement in any metric by
+    disabling.
```

Comment:
Honestly I was considering not documenting this at all but I feel equally icky about having undocumented settings. It's really there just so we can actually evaluate whether it even makes a difference or not. So far the answer seems "probably not" but it's less risky to leave it alone than change the behaviour since it doesn't help to remove it either.

Do you think this would be better:
 1. Undocumented (but still present for when we want to do perf testing)
 2. Documented as "don't use this it's just for internal testing"
 3. Documented as you proposed above, less verbose but with a general "you probably don' need to touch this ever" vibe?